src/callback.i

```diff
@@ -0,0 +1,45 @@
+// This enables the special std::string conversion maps
+// Without it, SWIG will wrap the C++ std::string as a special type
+%include <std_string.i>
+
+%include <std_map.i>
+
+%{
+// This goes into the wrapper code
+#include "callback.h"
+#include <vector>
+%}
+
+// This typemap converts a JS callback to a C++ std::function
+// This is the synchronous version, it creates a Local function reference (js_callback)
+// that exists for the duration of the call
+%typemap(in) std::function<std::string(int, const std::string &)> giver (Napi::Function js_callback) {
+  if (!$input.IsFunction()) {
+    %argument_fail(SWIG_TypeError, "$type", $symname, $argnum);
+  }
+  js_callback = $input.As<Napi::Function>();
+
+  // $1 is what we pass to the C++ function -> it is a C++ wrapper
+  // around the JS callback
+  $1 = [&js_callback, &env](int passcode, const std::string &name) -> std::string {
```

Comment:
`&env` seems ok in sync callbacks, but in an async callback, the `env` should be copied here to prevent further problems.
So is it a better way to also use copied `env` in a sync callback to keep consistent?

Reply:
There is much more to do to make it work in async, I will do it shortly